### PR TITLE
To add Nesterov Adam algorithm for multi-tensor optimizers API

### PR DESCRIPTION
**Pull request by iramazanli**
Previously in the PR: #59009 we added NAdam to Optimizers.  Here in this PR we are proposing multi-tensor version of NAdam for PyTorch.

Nadam has been proposed in the paper   https://openreview.net/forum?id=OM0jvwB8jIp57ZJjtNEZ and report  and report : http://cs229.stanford.edu/proj2015/054_report.pdf by Timothy Dozat.

It has been one of the most used algorithm in Deep Learning community.

It worth to noting that the implementation of NAdam is inspired by the implementation for Keras : 
https://github.com/tensorflow/tensorflow/blob/f9d386849581d15d72f6f1f96f12aac230a8edbe/tensorflow/python/keras/optimizer_v2/nadam.py

### CODE REVIEW

File: torch/optim/_functional.py [hunk 1]
```diff
@@ -357,3 +357,48 @@ def asgd(params: List[Tensor],
             ax.add_(param.sub(ax).mul(mu))
         else:
             ax.copy_(param)
+
+
+def nadam(params: List[Tensor],
```
Comment (vincentqb) [inline torch/optim/_functional.py:362]: does this belong to this PR? looks like #59009 got added here?

**Review by vincentqb — COMMENTED** [review]
Let's look at #59009 first

File: torch/optim/_functional.py (hunk 1 above)
Comment (iramazanli) [reply torch/optim/_functional.py]: it belongs to #59009   :

**Review by vincentqb — APPROVED** [review]
LGTM. please use `4e-5` for the tolerance on gpu so that the test passes.

**Pull request closed without merging**